datasets/lama/lama.py

```diff
@@ -0,0 +1,390 @@
+  title={Language Models as Knowledge Bases?},
+  author={F. Petroni, T. Rockt{\"{a}}schel, A. H. Miller, P. Lewis, A. Bakhtin, Y. Wu and S. Riedel},
+  booktitle={In: Proceedings of the 2019 Conference on Empirical Methods in Natural Language Processing (EMNLP), 2019},
+  year={2019}
+}
+
+@inproceedings{petroni2020how,
+  title={How Context Affects Language Models' Factual Predictions},
+  author={Fabio Petroni and Patrick Lewis and Aleksandra Piktus and Tim Rockt{\"a}schel and Yuxiang Wu and Alexander H. Miller and Sebastian Riedel},
+  booktitle={Automated Knowledge Base Construction},
+  year={2020},
+  url={https://openreview.net/forum?id=025X0zPfn}
+}
+"""
+
+
+_DESCRIPTION = """LAMA is a dataset used to probe and analyze the factual and commonsense knowledge contained in pretrained language models. See https://github.com/facebookresearch/LAMA.
+"""
+
+_HOMEPAGE = "https://github.com/facebookresearch/LAMA"
+
+_LICENSE = "The Creative Commons Attribution-Noncommercial 4.0 International License. see https://github.com/facebookresearch/LAMA/blob/master/LICENSE"
+
+_URLs = {
+    "trex": "https://dl.fbaipublicfiles.com/LAMA/negated_data.tar.gz",
+    "squad": "https://dl.fbaipublicfiles.com/LAMA/negated_data.tar.gz",
+    "google_re": "https://dl.fbaipublicfiles.com/LAMA/negated_data.tar.gz",
+    "conceptnet": "https://dl.fbaipublicfiles.com/LAMA/negated_data.tar.gz",
+}
+
+
+class Lama(datasets.GeneratorBasedBuilder):
+    """Lama Dataset"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="trex", version=VERSION, description="The TRex part of the Lama dataset"),
+        datasets.BuilderConfig(name="squad", version=VERSION, description="The Squad part of the Lama dataset"),
+        datasets.BuilderConfig(
+            name="google_re", version=VERSION, description="The Google_re part of the Lama dataset"
+        ),
+        datasets.BuilderConfig(
+            name="conceptnet", version=VERSION, description="The Conceptnet part of the Lama dataset"
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "trex"
+
+    def _info(self):
+        if self.config.name == "trex":
+            features = datasets.Features(
+                {
+                    "uuid": datasets.Value("string"),
+                    "obj_uri": datasets.Value("string"),
+                    "obj_label": datasets.Value("string"),
+                    "sub_uri": datasets.Value("string"),
+                    "sub_label": datasets.Value("string"),
+                    "predicate_id": datasets.Value("string"),
+                    "sub_surface": datasets.Value("string"),
+                    "obj_surface": datasets.Value("string"),
+                    "masked_sentence": datasets.Value("string"),
+                    "template": datasets.Value("string"),
+                    "template_negated": datasets.Value("string"),
+                    "label": datasets.Value("string"),
```

Comment:
Would it be possible to use the ClassLabel feature type here ?
What values the `label` field can take ?

Reply:
Contrary to the name of the field by the author, I don't consider this a class label: These are the values of the field, so I prefer not to make it a classlabel. We can name the field something else to minimize confusion?

['applies to jurisdiction', 'field of work', 'native language', 'occupation', 'employer', 'owned by', 'instrument', 'located in the administrative territorial entity', 'genre', 'capital of', 'named after', 'religion', 'languages spoken, written or signed', 'headquarters location', 'country', 'manufacturer', 'developer', 'place of birth', 'twinned administrative body', 'place of death', 'record label', 'country of citizenship', 'location', 'subclass of', 'continent', 'instance of', 'capital', 'part of', 'original language of film or TV show', 'official language', 'position held', 'language of work or name', 'position played on team / speciality', 'original network', 'member of', 'shares border with', 'country of origin', 'has part', 'diplomatic relation', 'location of formation', 'work location']

Reply:
We can always make it a classlabel later if users think it's useful.

Reply:
Indeed it's clearly not the main objective of this dataset.
We can keep it this way